.jenkins/Jenkinsfile

```diff
@@ -0,0 +1,160 @@
+  ]);
+}
+
+pipeline {
+    agent any
+    tools {
+        maven 'Maven 3.8.1'
+    }
+    stages {
+        stage('Set build status on GitHub') {
+            steps {
+                setBuildStatus 'Build #${env.BUILD_NUMBER} has started...', 'PENDING'
+            }
+        }
+        stage('Checkout scm') {
+            steps {
+                checkout scm
+                script {
+                    def pom = readMavenPom();
+                    testdbName = pom.properties['gemma.testdb.name'];
+                    gemmaVersion = pom.version;
+                    if (env.BRANCH_NAME == "master" && gemmaVersion.endsWith("-SNAPSHOT")) {
+                        error("A production build must not have a -SNAPSHOT suffix.");
+                    }
+                    if (env.BRANCH_NAME == "development" && !gemmaVersion.endsWith("-SNAPSHOT")) {
+                        error("A development build must have a -SNAPSHOT suffix.");
+                    }
+                    for (dep in pom.dependencies) {
+                        if (dep.groupId == 'baseCode' && dep.artifactId == 'baseCode') {
+                            baseCodeVersion = dep.version;
+                            break;
+                        }
+                    }
+                    if (baseCodeVersion == null) {
+                        error("The baseCode dependency was not found in pom.xml.");
+                    }
+                    if (env.BRANCH_NAME == 'development') {
+                        dataDir = '/space/sandboxgemmaData';
+                    }
+                    if (env.BRANCH_NAME =~ '^(hotfix|release)-.*') {
+                        dataDir = '/space/staginggemmaData';
+                    }
+                    if (env.BRANCH_NAME == 'master') {
+                        deployDir = 'moe:' + deployDir;
+                    } else {
+                        deployDir = 'chalmers:' + deployDir;
+                    }
+                    if (env.BRANCH_NAME =~ '^(hotfix|release)-.*') {
+                        deployDir += '-staging';
+                    }
+                    if (env.BRANCH_NAME != 'master') {
+                        cliDir += '-' + gemmaVersion;
+                    }
+                }
+            }
+        }
+        stage('Build') {
+            steps {
+                sh 'mvn -B package -DskipTests'
+            }
+        }
+        stage('Run quick unit tests') {
+            steps {
+                sh 'mvn -B test -DexcludedGroups=SlowTest'
+            }
+        }
+        stage('Run slow unit tests and integration tests') {
+            when {
+                branch 'hotfix-*';
+                branch 'release-*';
+                branch 'development'
+            }
+            steps {
+                sh 'mvn -B test -Dgroups=SlowTest'
+                lock('test-database/' + testdbName) {
```

Comment:
This could be simplified to `database/${testDbName}` since we don't use the database on chalmers anymore for testing.